lib/datadog/tracing.rb

```diff
@@ -86,54 +87,54 @@ def configuration
       def configure
         # Wrap block with trace option validation
         wrapped_block = proc do |c|
-          yield(Datadog::Core::Configuration::ValidationProxy::Tracing.new(c))
+          yield(Configuration::ValidationProxy.new(c))
         end
 
         # Configure application normally
         Datadog.send(:internal_configure, &wrapped_block)
       end
 
-      # (see Datadog::Tracer#active_trace)
+      # (see Datadog::Tracing::Tracer#active_trace)
       # @public_api
-      def active_trace
+      def active_trace(key: nil)
```

Comment:
I don't think we should expose the `key` argument in the public API.

There's no valid use case for a `ddtrace` user that I can think of that can be achieve by providing a custom key here.

Key is a misnomer, as it is actually the storage vessel for the context (we only use it with a `Thread` object ourselves).

I think we should hide this parameter unless there's a clear need for it.

Same applies for other methods in this file with the `key` parameter added in this changeset.

Reply:
There's unfortunately a clear need for it: profiling. It needs to access spans/traces per thread so it can get correlation information. It can only do so from its own thread too.